src/rounded_rect.rs

```diff
@@ -0,0 +1,138 @@
+//! A rectangle with rounded corners.
+
+use crate::{PathEl, Rect, Shape, Vec2};
+use std::f64::consts::PI;
+
+/// A rectangle with rounded corners.
+#[derive(Clone, Copy, Default, Debug)]
+pub struct RoundedRect {
+    /// Coordinates of the rectangle.
+    pub rect: Rect,
+    /// Radius of all four corners.
+    pub radius: f64,
```

Comment:
An interesting question, should there be a single radius value or one for each corner? I think I would lean toward the latter as it's more general and I can easily imagine such things used in UI.

Reply:
Good point, I currently focused on one radius for all corners as this is natively supported by d2d (actually they use ellipses but not sure how useful that is). Nonetheless, the backend could decide internally which path to use for rendering.